fastapi/fastapi_dispatcher.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
+
+from contextlib import contextmanager
+from io import BytesIO
+
+from odoo.http import Dispatcher, request
+
+from .context import odoo_env_ctx
+
+
+class FastApiDispatcher(Dispatcher):
+    routing_type = "fastapi"
+
+    @classmethod
+    def is_compatible_with(cls, request):
+        return True
+
+    def dispatch(self, endpoint, args):
+        # don't parse the httprequest let starlette parse the stream
+        self.request.params = {}  # dict(self.request.get_http_params(), **args)
+        environ = self._get_environ()
+        root_path = "/" + environ["PATH_INFO"].split("/")[1]
```

Comment:
```suggestion
        root_path = "/".join(environ["PATH_INFO"].split("/")[0:-1])
```
If your endpoint is like `api/v1` the router does not solve the path